backend/pkg/factory/kafka/client_hooks.go

```diff
@@ -130,3 +152,17 @@ func (c clientHooks) OnBrokerWrite(_ kgo.BrokerMetadata, _ int16, bytesWritten i
 	c.requestSentCount.Inc()
 	c.bytesSent.Add(float64(bytesWritten))
 }
+
+// IncrementClientCount increments the active client count metric
+func IncrementClientCount() {
+	if promActiveClients != nil {
+		promActiveClients.Inc()
+	}
+}
+
+// DecrementClientCount decrements the active client count metric
+func DecrementClientCount() {
+	if promActiveClients != nil {
+		promActiveClients.Dec()
+	}
+}
```

Comment:
Do we need those nil checks and therefore the wrapper functions rather than calling Inc() Dec() directly on the prometheus metrics like we do with the other hooks?

Unless you create hooks without using the constructor it should be impossible that the prom metrics are not initialized yet.

Reply:
I think this specific factory implementation only ever creates a single client so that's fine - but the other factory implementations (closed source) it's more dynamic.